build.sh

```diff
@@ -16,13 +16,140 @@
 
```

Comment:
At a certain point I must ask - would this bash script be more readable/maintainable as a Python script instead? It seems roughly half of this could be replaced by a few lines of `argparse`.